pandas/core/indexes/datetimelike.py

```diff
@@ -508,16 +508,16 @@ def _wrap_range_setop(self, other, res_i8):
         result = type(self._data)._simple_new(
             res_values, dtype=self.dtype, freq=new_freq
         )
-        return self._wrap_setop_result(other, result)
+        return cast("Self", self._wrap_setop_result(other, result))
```

Comment:
Looks like `Index._wrap_setop_result` should return `Self`instead of `Index`? Then you could avoid this cast?

Reply:
Can't bc it uses shallow_copy which doesn't return Self for RangeIndex